eclass/blas.eclass

```diff
@@ -0,0 +1,241 @@
+# @ECLASS: blas.eclass
+# @MAINTAINER:
+# schmidom@student.ethz.ch
+# @BLURB: Selects a proper BLAS implementation to build the package
+# @DESCRIPTION:
+# Here a BLAS-implementation is chosen from the cutset of available,
+# compatible and implementations in the USE-flags
+# The implementation is then enforced upon every dependency of the package
+
+# @ECLASS-VARIABLE: BLAS_COMPAT
+# @REQUIRED
+# @DESCRIPTION:
+# This variable contains a list of BLAS implementations the package
+# supports. It has to be an array. Either it or BLAS_COMPAT_ALL must be
+# set before the `inherit' call. BLAS_COMPAT_ALL overrides BLAS_COMPAT.
+#
+# Example:
+# @CODE
+# BLAS_COMPAT=( refblas openblas gotoblas mkl )
+# @CODE
+
+# @ECLASS-VARIABLE: BLAS_COMPAT_ALL
+# @REQUIRED
+# @DESCRIPTION:
+# This variable marks that the package is compatible with all (standard)
+# BLAS implementations. Set it to a non-empty value to make your package
+# compatible with any BLAS version. Either it or BLAS_COMPAT must be set
+# before the `inherit' call. BLAS_COMPAT_ALL overrides BLAS_COMPAT.
+#
+# Example:
+# @CODE
+# BLAS_COMPAT_ALL=1
+# @CODE
+
+# @ECLASS-VARIABLE: BLAS_USEDEP
+# @DESCRIPTION:
+# This is an eclass-generated USE-dependency string which has to be used
+# to propagate the chosen BLAS-implementation down the dependency graph
+#
+# Example use:
+# @CODE
+# RDEPEND="sci-libs/foo[${BLAS_USEDEP}]"
+# @CODE
+#
+# Example value:
+# @CODE
+# refblas?,openblas?
+# @CODE
+
+
+# @ECLASS-VARIABLE: BLAS_CONDITIONAL_FLAG
```

Comment:
1. This is a horrible trap, and you already fell in it. Again, python-r1 doesn't do stuff like this for a good reason. You add the basic form and it works for 95% of the packages. Then you add more complexity and it works for 95% of the rest. Then you…
2. Implicit OR is totally confusing. What if someone needs AND instead?

Reply:
What would be the proper way to handle it? define a BLAS_DEPEND that the eclass-dev then can add to DEPEND themselves?